### PR TITLE
Added advanced topic section for contributors

### PR DESCRIPTION
**Pull request by eldadfux**
## What does this PR do?

Added a new advanced topic section for contributors who wish to better understand the Appwrite design and codebase.

## Test Plan

N/A

## Related PRs and Issues

N/A

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yes

### CODE REVIEW

**Comment by eldadfux** [comment]
Please suggest other topics you think may be useful.

**Comment by TorstenDittmann** [comment]
We could add https://en.wikipedia.org/wiki/OAuth as well 👍🏻

**Comment by eldadfux** [comment]
> We could add https://en.wikipedia.org/wiki/OAuth as well 👍🏻

good one...

**Review by TorstenDittmann — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -90,6 +90,34 @@ Appwrite uses an internal micro-framework called Litespeed.js to build simple UI
 
 After finishing the installation process, you can start writing and editing code.
 
+
+#### Advanced Topics
+We love to create issues that are good for begginers and label them as `good for begginers` or `hacktoberfest`, but some more advanced topics might require extra knowledge. Below is a list with links for which you can use to learn more about some of them more advance topics that will help you master the Appwrite codebase.
```
Comment (lohanidamodar) [inline CONTRIBUTING.md:95]: *beginners not begginers

* Below is a list **with links which** you can use to learn more about some **of the more** advance topics that will help you master Appwrite codebase.

**Review by lohanidamodar — CHANGES_REQUESTED** [review]

**Comment by kodumbeats** [comment]
Maybe some literature about the frontend console? Possible suggestions:
- [MVVM](https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93viewmodel)
- [Gulp](https://gulpjs.com/docs/en/getting-started/quick-start)

**Comment by eldadfux** [comment]
Adapted the list according to all comments, feel free to submit new stuff at any time.

**Pull request merged**